projects/sound-open-firmware/Dockerfile

```
@@ -20,39 +20,57 @@
 
 FROM gcr.io/oss-fuzz-base/base-builder
 
-RUN apt-get -y update && \
-	    apt-get install -y \
-	    	autoconf \
-		bison \
-		build-essential \
-		flex \
-		gawk \
-		gettext \
-		git \
-		gperf \
-		help2man \
-		libncurses5-dev \
-		libssl-dev \
-		libtool \
-		libtool-bin \
-		pkg-config \
-		software-properties-common \
-		texinfo \
-		udev \
-		cmake \
-		libglib2.0-dev
-
-ARG CLONE_DEFAULTS="--depth 5"
-
-# Use ToT alsa utils for the latest topology patches.
-RUN cd /tmp && \
-git clone $CLONE_DEFAULTS https://github.com/thesofproject/alsa-lib.git && \
-git clone $CLONE_DEFAULTS https://github.com/thesofproject/alsa-utils.git && \
-cd /tmp/alsa-lib && ./gitcompile &&  make install && \
-cd /tmp/alsa-utils && ./gitcompile &&  make install
-
-ARG GITHUB_SOF=https://github.com/thesofproject
-
-RUN cd $SRC && git clone $CLONE_DEFAULTS $GITHUB_SOF/sof
-WORKDIR sof
+# Base packages
+RUN apt-get -y update
+RUN apt-get install -y \
+      gettext git libc6-dev-i386 libglib2.0-dev libncurses5-dev \
+      libtool ninja-build python3-pip
+RUN pip3 install west
+
+# Zephyr SDK:
+#
+# Zephyr doesn't provide a clean "get the latest version" URL, but
+# note that the use of the /latest/ path component at least ensures
+# this will fail on a new release.
+ARG SDK_VER=0.16.1
+WORKDIR /root
+RUN curl -L https://github.com/zephyrproject-rtos/sdk-ng/releases/latest/download/zephyr-sdk-${SDK_VER}_linux-x86_64_minimal.tar.xz \
+    | tar xfJ -
+RUN zephyr-sdk-*/setup.sh -h
+
+WORKDIR $SRC
+
+# ALSA: (SOF needs tracks upstream closely, but doesn't yet pull these
+# via west or track commits internally)
+RUN git clone https://github.com/thesofproject/alsa-lib.git
+RUN git clone https://github.com/thesofproject/alsa-utils.git
+RUN cd $SRC/alsa-lib && ./gitcompile && make install
+RUN cd $SRC/alsa-utils && ./gitcompile && make install
+
+# SOF itself (also pulls in Zephyr + modules via west)
+RUN west init -m https://github.com/thesofproject/sof sof
+RUN cd sof && west update
```

You will get a 2x speed up with this recent and wonderful git option. I started carpet bombing SOF CI with it and it's been fantastic

```suggestion
RUN cd sof && west update --fetch-opt=--filter=tree:0
```

It's better than --depth because it breaks neither git describe nor git merge-base https://github.com/thesofproject/linux/issues/2556

More info inhttps://github.com/zephyrproject-rtos/west/issues/638 and others

---

Hm... getting `fatal: --filter can only be used when extensions.partialClone is set` when I try this?  Maybe this is too bleeding edge for the git in the 20.04 image?  Or maybe I just need to read some more docs.  CI optimization is pretty far from my happy zone.

---

> Maybe this is too bleeding edge for the git in the 20.04 image? 

Yes: I can reproduce with git 2.25.1 in Ubuntu 20.04 but not with git 2.34.1 in Ubuntu 22.04.

This happens only when trying to fetch with the `--filter` optimization _on an existing, not already "filtered" clone_ (and only on 20.04).

This happens because west clones missing modules in two steps: 1. git initialization of a "non-filtered". 2. git fetch --filter=...  (boom)

A tested workaround is `git config remote.upstream.partialclonefilter tree:0` between 1. and 2. but of course you can't inject any command between west steps 1. and 2.

https://git-scm.com/docs/repository-version

Ubuntu 22.04 is not affected because git fetch with version 2.34.1 automagically adds the `git config` above.

Do you really need to use Ubuntu 20.04? We recently upgraded everything to 22.04 and we experienced a few bugs but absolutely nothing related to toolchains or building anything.

---

> You will get a 2x speed up with this recent and wonderful git option.

Finally woke up and realized this is run _at image creation time_; to sort-of "warm up a git cache". So while this git optimization trick is very good to know it's totally off-topic here, sorry the distraction. Resolving this comment.

---

@marc-hb its because we are not part of the repo